ext/storage_sources/gcp_store/gcp_connection.cpp

```diff
@@ -63,8 +63,19 @@ gcp_connection::list_objects(std::vector<std::string> &objects, bool list_single
 
 // Puts an object into a google cloud bucket.
 int
-gcp_connection::put_object(const std::string &object_key, const std::string &file_name) const
+gcp_connection::put_object(const std::string &object_key, const std::string &file_path)
 {
+    // Note that the client library automatically computes a hash on the
+    // client-side to verify data integrity during transmission.
+    google::cloud::v2_5_0::StatusOr<gcs::ObjectMetadata> metadata =
```

Comment:
The ticket talks about an `IfGenerationMatch(0)` precondition that I don't see here. Is it no longer required?

Reply:
Also on the nitpicky side `google::cloud::v2_5_0::` feels like something we should be namespacing so there's only a single occurrence of `v2_5_0` in the codebase in case we decide to update versions later. 
Ping @tammybailey for a second opinion.

Reply:
IfGenerationMatch(0) is no longer needed since we check if the file exists before we pass in the filename. So I removed it here for optimisation. 

Currently this is the only occurrence of v2_5_0, it might be used again when I implement bucket_exists so I'm leaning towards namespacing it.

Reply:
It seems like there shouldn't be a version number in here at all. The documentation uses `google::cloud::StatusOr` throughout. Can you point me to the reference that specifies the SDK version as part of the namespace?

Also, the latest version is 2.6. Is there a reason you are using 2.5?

Reply:
Initially it was underlined as an error when I use `google::cloud::StatusOr`, so I followed what we did in the example code to add v2_5_0 which resolved the error. However when I tested it just now, it worked with `google::cloud::StatusOr`. So I'm happy to remove v2_5_0 for my next commit. :))

Reply:
Thanks @bzm-jas! Would you mind updating the Jira comment with these `IfGenerationMatch(0)` details?

We've defined `_gcp_client` as `v2_5_0` in [gcp_connection.h](https://github.com/wiredtiger/wiredtiger/blob/f1ca320e79c7811223427076bfc963492b589614/ext/storage_sources/gcp_store/gcp_connection.h) which would explain why you got the type warning. It's worth discussing why we have 2.5 instead of 2.6, but I think we can handle that outside this PR.

Reply:
I'm fairly certain that `gcp_client` shouldn't be defined that way either. This should be correct:
```
google::cloud::storage::Client _gcp_client;
```

Reply:
version number has been removed here: [33a5430](https://github.com/wiredtiger/wiredtiger/pull/8730/commits/33a543090bfa86eeca8009faca848594c309f792) 😊